lib/_http_outgoing.js

```diff
@@ -365,14 +372,16 @@ OutgoingMessage.prototype._renderHeaders = function() {
     throw new Error('Can\'t render headers after they are sent to the client.');
   }
 
-  if (!this._headers) return {};
+  var _headers = this._headers;
+  if (!_headers) return {};
 
   var headers = {};
   var keys = Object.keys(this._headers);
```

Comment:
Use `_headers` (or whatever better name you can come up with :-) here as well for consistency.